modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/servlet/taglib/GroupSelectorTag.java

```diff
@@ -128,14 +127,21 @@ private List<Group> _getGroups(HttpServletRequest httpServletRequest) {
 			int[] startAndEnd = SearchPaginationUtil.calculateStartAndEnd(
 				cur, delta);
 
-			_groups = groupItemSelectorProviderOptional.map(
-				groupItemSelectorProvider ->
-					groupItemSelectorProvider.getGroups(
-						group.getCompanyId(), group.getGroupId(), keywords,
-						startAndEnd[0], startAndEnd[1])
-			).orElse(
-				Collections.emptyList()
-			);
+			if (groupItemSelectorProvider == null) {
```

Comment:
@JanisZhang Please SF this method and `_getGroupsCount` to apply "shortcut" logic.

For example, you can `return _groups` in the `if` at line 111, so you don't need to indent for the `else {` part; then you can also shortcut in this `if`, etc...